ProcessLib/RichardsFlow/RichardsFlowFEM.h

```diff
@@ -0,0 +1,193 @@
+
+public:
+    LocalAssemblerData(MeshLib::Element const& element,
+                       std::size_t const local_matrix_size,
+                       bool is_axially_symmetric,
+                       unsigned const integration_order,
+                       RichardsFlowProcessData const& process_data)
+        : _element(element),
+          _process_data(process_data),
+          _integration_method(integration_order),
+          _shape_matrices(initShapeMatrices<ShapeFunction, ShapeMatricesType,
+                                            IntegrationMethod, GlobalDim>(
+              element, is_axially_symmetric, _integration_method)),
+          _saturation(
+              std::vector<double>(_integration_method.getNumberOfPoints()))
+    {
+    }
+
+    void assemble(double const t, std::vector<double> const& local_x,
+                  std::vector<double>& local_M_data,
+                  std::vector<double>& local_K_data,
+                  std::vector<double>& local_b_data) override
+    {
+        auto const local_matrix_size = local_x.size();
+        // This assertion is valid only if all nodal d.o.f. use the same shape
+        // matrices.
+        assert(local_matrix_size == ShapeFunction::NPOINTS * NUM_NODAL_DOF);
+
+        MathLib::PiecewiseLinearInterpolation const& interpolated_Pc =
+            *_process_data.curves.at("curveA");
+        MathLib::PiecewiseLinearInterpolation const& interpolated_Kr =
+            *_process_data.curves.at("curveB");
+
+        auto local_M = MathLib::createZeroedMatrix<NodalMatrixType>(
+            local_M_data, local_matrix_size, local_matrix_size);
+        auto local_K = MathLib::createZeroedMatrix<NodalMatrixType>(
+            local_K_data, local_matrix_size, local_matrix_size);
+        auto local_b = MathLib::createZeroedVector<NodalVectorType>(
+            local_b_data, local_matrix_size);
+
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+        SpatialPosition pos;
+        pos.setElementID(_element.getID());
+        double P_int_pt = 0.0;
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            pos.setIntegrationPoint(ip);
+            auto const& sm = _shape_matrices[ip];
+            auto const& wp = _integration_method.getWeightedPoint(ip);
+
+            NumLib::shapeFunctionInterpolate(local_x, sm.N, P_int_pt);
+
+            auto const K = _process_data.intrinsic_permeability(t, pos)[0];
+            auto const poro = _process_data.porosity(t, pos)[0];
+            auto const mu = _process_data.viscosity(t, pos)[0];
+            auto const storage = _process_data.storage(t, pos)[0];
+            auto const rho_w = _process_data.water_density(t, pos)[0];
+            auto const body_force = _process_data.specific_body_force(t, pos);
+            assert(body_force.size() == GlobalDim);
+            auto const b =
```

Comment:
You can use MathLib::toVector as
```
 auto const& b = MathLib::toVector(body_force, GlobalDim);
```
with including #include "MathLib/LinAlg/Eigen/EigenMapTools.h"